### PR TITLE
feat(snapchat_conversion): add products to contents mapping

### PR DESCRIPTION
**Pull request by maheshkutty**
## What are the changes introduced in this PR?

This pull request enhances the Snapchat Conversion destination integration by introducing a new utility function to standardize product-to-contents mapping and updating the transformation logic and tests to support this feature. The main goal is to ensure the `contents` field in outgoing payloads is consistently populated based on product data, improving downstream analytics and event tracking.

**Transformation logic improvements:**

* Added the `productsToContentsMapping` utility function to `utilsV3.js`, which extracts and normalizes product information from incoming messages to generate a standardized `contents` array for Snapchat payloads. This function handles various edge cases, such as missing or malformed product data, and ensures only valid content objects are included.
* Updated `transformV3.js` to use the new `productsToContentsMapping` function within the payload builder, ensuring the `contents` field is added to outgoing payloads when applicable. [[1]](diffhunk://#diff-8ddcbc518509c68b87f33c9e21799d63d66b92f25ec99a2c704e38d11eae35b6L34-R34) [[2]](diffhunk://#diff-8ddcbc518509c68b87f33c9e21799d63d66b92f25ec99a2c704e38d11eae35b6R168-R171)

**Testing and validation:**

* Added comprehensive unit tests for the `productsToContentsMapping` function in `utilsV3.test.js`, covering a wide range of input scenarios and edge cases to guarantee correct behavior.
* Updated processor test data in `dataV3.ts` to include the new `contents` field in expected outputs, ensuring integration tests validate the correct population of product details for all relevant event types. [[1]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R864-R874) [[2]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R1041-R1051) [[3]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R1219-R1230) [[4]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R1389-R1394) [[5]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R1531-R1537) [[6]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R1673-R1679) [[7]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R1938-R1942) [[8]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R2075-R2079) [[9]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R2212-R2216) [[10]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R2349-R2353) [[11]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R2487-R2491) [[12]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R2626-R2630) [[13]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R2918-R2923) [[14]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R3488-R3499) [[15]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R3597-R3598) [[16]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R3668-R3679) [[17]](diffhunk://#diff-001eef54c6eb559f96974d10675f11ad10ce1ec9af998aca7296ba95b1545af0R3780)

**Utility enhancements:**

* Extended imports from the shared `util` module in `utilsV3.js` to leverage helper functions for object validation and cleanup, supporting the new mapping logic.

These changes collectively improve the reliability and consistency of product data handling for Snapchat Conversion events.

## What is the related Linear task?

Resolves INT-3958

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4526/index.html)

**Review by ItsSudip — DISMISSED** [review]

File: src/v0/destinations/snapchat_conversion/utilsV3.js [hunk 1]
```diff
@@ -131,7 +131,21 @@ const getExtInfo = (message) => {
   return extInfo.map((value) => (value == null ? '' : value));
 };
 
+const productsToContentsMapping = (message) => {
```
[the remark this answers is not in the document]
Comment (maheshkutty) [reply src/v0/destinations/snapchat_conversion/utilsV3.js:134]: I am only adding for those events which contains products
https://github.com/rudderlabs/rudder-transformer/pull/4526/files#diff-8ddcbc518509c68b87f33c9e21799d63d66b92f25ec99a2c704e38d11eae35b6R148

**Comment by devops-github-rudderstack** [comment]
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4526/index.html)

**Comment by devops-github-rudderstack** [comment]
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4526/index.html)
